#Import Libraries

In [ ]:
import pandas as pd
import numpy as np

#Load the dataset with pandas.read_csv() and inspect it using .info(), .describe(), and .head()

In [2]:
df=pd.read_csv('/content/Titanic Dataset.csv')

In [3]:
df.head(5)

,PassengerId,Survived
0,892,0
1,893,1
2,894,0
3,895,0
4,896,1


In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 2 columns):
 #   Column       Non-Null Count  Dtype
---  ------       --------------  -----
 0   PassengerId  418 non-null    int64
 1   Survived     418 non-null    int64
dtypes: int64(2)
memory usage: 6.7 KB


In [9]:
df.describe()

,PassengerId,Survived
count,418.000000,418.000000
mean,1100.500000,0.363636
std,120.810458,0.481622
min,892.000000,0.000000
25%,996.250000,0.000000
50%,1100.500000,0.000000
75%,1204.750000,1.000000
max,1309.000000,1.000000


#**Identify**: how many rows/columns, which columns have missing values, and which are categorical vs numerical

In [5]:
df.shape

(418, 2)

In [6]:
df.isnull().sum()

,0
PassengerId,0
Survived,0


In [10]:
print("--- DATA TYPE BREAKDOWN ---")
numerical_cols = df.select_dtypes(include=np.number).columns.tolist()
categorical_cols = df.select_dtypes(include='object').columns.tolist()

print(f"Numerical Columns ({len(numerical_cols)}):\n{numerical_cols}\n")
print(f"Categorical Columns ({len(categorical_cols)}):\n{categorical_cols}")

--- DATA TYPE BREAKDOWN ---
Numerical Columns (2):
['PassengerId', 'Survived']

Categorical Columns (0):
[]


In [11]:
print("--- STATISTICAL SUMMARY ---")
df.describe()

--- STATISTICAL SUMMARY ---


,PassengerId,Survived
count,418.000000,418.000000
mean,1100.500000,0.363636
std,120.810458,0.481622
min,892.000000,0.000000
25%,996.250000,0.000000
50%,1100.500000,0.000000
75%,1204.750000,1.000000
max,1309.000000,1.000000


### Initial Data Story

The dataset consists of 418 entries and 2 columns: 'PassengerId' and 'Survived'. Both columns are of integer type (`int64`). Importantly, there are no missing values in the dataset, which simplifies the initial data cleaning process. The 'Survived' column appears to be a binary indicator (0 or 1), likely representing whether a passenger survived or not.